armi/reactor/converters/uniformMesh.py

```diff
@@ -171,18 +222,66 @@ def applyStateToOriginal(self):
-        self._clearStateOnReactor(self._sourceReactor, cache=True)
-        self._mapStateFromReactorToOther(self.convReactor, self._sourceReactor)
+        # If we have non-uniform mesh assemblies then we need to apply a
+        # different approach to undo the geometry transformations on an
+        # assembly by assembly basis.
+        if self._hasNonUniformAssems:
+            for assem in self._sourceReactor.core.getAssemblies(
+                self._nonUniformMeshFlags
+            ):
+                for storedAssem in self._sourceReactor.core.sfp.getChildren():
+                    if (
+                        storedAssem.getName()
+                        == assem.getName() + self._SFP_TEMP_STORAGE_SUFFIX
+                    ):
+                        self.setAssemblyStateFromOverlaps(
+                            assem,
+                            storedAssem,
+                            self.blockParamNames,
+                            mapNumberDensities=False,
+                        )
+
+                        # Remove the stored assembly from the spent fuel pool
+                        # and replace the current assembly with it.
+                        storedAssem.spatialLocator = assem.spatialLocator
+                        storedAssem.setName(assem.getName())
+                        if (
+                            storedAssem.spatialLocator
+                            in self._sourceReactor.core.sfp.childrenByLocator
+                        ):
+                            self._sourceReactor.core.sfp.childrenByLocator.pop(
+                                storedAssem.spatialLocator
+                            )
+
+                        self._sourceReactor.core.sfp.remove(storedAssem)
+                        self._sourceReactor.core.removeAssembly(assem, discharge=False)
+                        self._sourceReactor.core.add(storedAssem)
+                        break
+                else:
+                    runLog.error(
+                        f"No assembly matching name {assem.getName()} "
+                        f"was found in the spent fuel pool. {assem} "
+                        f"will persist as an axially unified assembly. "
+                        f"This is likely not intended."
+                    )
+
+            self._sourceReactor.core.updateAxialMesh()
+        else:
+            # Clear the state of the original source reactor to ensure that
+            # a clean mapping between the converted reactor for data that has been
+            # changed. In this case, we cache the original reactor's data so that
+            # after the mapping has been applied, we can recover data from any
+            # parameters that did not change.
+            self._cachedReactorCoreParamData = {}
+            self._clearStateOnReactor(self._sourceReactor, cache=True)
+            self._mapStateFromReactorToOther(
+                self.convReactor, self._sourceReactor, mapNumberDensities=False
```

Comment:
To maintain the previous behavior here, shouldn't `mapNumberDensities = True`?

Reply:
At least that is the assumption here. Maybe it's "safer" right now just to keep the same behavior so we don't introduce some weird bug...

Reply:
Okay updated back to how it was.

Reply:
https://github.com/terrapower/armi/pull/879/commits/522d885ddf78aa7ce9224cb9f2c0ce18212cab47

If you believe that it's not needed, let's make an issue for it and get it in a subsequent PR 👍